tests/test_freqtradebot.py

```diff
@@ -3622,85 +3648,29 @@ def test_get_real_amount_multi(default_conf, trades_for_order2, buy_order_fee, c
         open_rate=0.245441,
         open_order_id="123456"
     )
-    freqtrade = get_patched_freqtradebot(mocker, default_conf)
-
-    # Amount is reduced by "fee"
-    assert freqtrade.get_real_amount(trade, buy_order_fee) == amount - (amount * 0.001)
-    assert log_has('Applying fee on amount for Trade(id=None, pair=LTC/ETH, amount=8.00000000, '
-                   'open_rate=0.24544100, open_since=closed) (from 8.0 to 7.992).',
-                   caplog)
-
-    assert trade.fee_open == 0.001
-    assert trade.fee_close == 0.001
-    assert trade.fee_open_cost is not None
-    assert trade.fee_open_currency is not None
-    assert trade.fee_close_cost is None
-    assert trade.fee_close_currency is None
 
-
-def test_get_real_amount_multi2(default_conf, trades_for_order3, buy_order_fee, caplog, fee,
-                                mocker, markets):
-    # Different fee currency on both trades
-    mocker.patch('freqtrade.exchange.Exchange.get_trades_for_order', return_value=trades_for_order3)
-    amount = float(sum(x['amount'] for x in trades_for_order3))
-    default_conf['stake_currency'] = 'ETH'
-    trade = Trade(
-        pair='LTC/ETH',
-        amount=amount,
-        exchange='binance',
-        fee_open=fee.return_value,
-        fee_close=fee.return_value,
-        open_rate=0.245441,
-        open_order_id="123456"
-    )
     # Fake markets entry to enable fee parsing
     markets['BNB/ETH'] = markets['ETH/BTC']
     freqtrade = get_patched_freqtradebot(mocker, default_conf)
     mocker.patch('freqtrade.exchange.Exchange.markets', PropertyMock(return_value=markets))
-    mocker.patch('freqtrade.exchange.Exchange.fetch_ticker',
-                 return_value={'ask': 0.19, 'last': 0.2})
+    mocker.patch(
+        'freqtrade.exchange.Exchange.fetch_ticker',
+        return_value={'ask': 0.19, 'last': 0.2}
+    )
 
     # Amount is reduced by "fee"
-    assert freqtrade.get_real_amount(trade, buy_order_fee) == amount - (amount * 0.0005)
```

Comment:
You'll have to look at `trades_for_order3` (probably in `conftest.py`) for this.
This test is about gathering fees from a trade-result (because `fetch_order()` didn't contain fees) - and there were 2 trade necessary to fill your order.

I'm not 100% sure if combining all of them into one makes sense (must look at the details of how it's done), as the mocks will need to be different (return something different).

Reply:
I actually meant to remove that fixture, it is only used in this test case, is built on top of the `trades_for_order2` and the fixture only changes the `trades_for_order2` so slightly that it's probably worth just setting as a parameter. Unless we prefer to keep these in fixtures, in which case parameterizing probably doesn't make sense.

https://github.com/freqtrade/freqtrade/blob/6fc770d97dd8d4380be3fb94e8f3dcedfa981cd7/freqtrade/resolvers/conftest.py

Reply:
Removed that fixture